mam/src/mam.rs

```diff
@@ -165,41 +195,177 @@ mod tests {
             .flat_map(char_to_trits)
             .cloned()
             .collect();
-        let message: Vec<Trit> = "IAMSOMEMESSAGE9HEARMEROARMYMESSAGETOTHEWORLDYOUHEATHEN"
+        let message: Vec<Trit> = "ABCDEFGHIJKLMNOPQRSTUVWXYZ9ABCDEFGHIJKLMNOPQRSTUVWXYZ9"
+            .chars()
+            .flat_map(char_to_trits)
+            .cloned()
+            .collect();
+        let side_key: Vec<Trit> = "EFGHIJKLMNOPQRSTUVWXYZ9ABCDEFGHIJKLMNOPQRSTUVWXYZ9ABCDABCD"
             .chars()
             .flat_map(char_to_trits)
             .cloned()
             .collect();
         let security = 1;
         let start = 1;
-        let count = 9;
+        let count = 4;
         let next_start = start + count;
-        let next_count = 4;
-        let index = 3;
+        let next_count = 2;
+        let index = 1;
 
         let mut c1 = CpuCurl::<Trit>::default();
         let mut c2 = CpuCurl::<Trit>::default();
         let mut bc = CpuCurl::<BCTrit>::default();
 
-        let (masked_payload, root) = create::<CpuCurl<Trit>, CpuCurl<BCTrit>, CpuHam>(
+        let mut key: Vec<Trit> = vec![0 as Trit; security as usize * iss::KEY_LENGTH];
+        let (siblings, root) = {
+            let mut digest = [0 as Trit; iss::DIGEST_LENGTH];
+
+            let addresses: Vec<Vec<Trit>> = {
```

Comment:
Could we move this to helper methods? People will need to run this often enough and shouldn't need to understand our signing scheme that much to be able to do so.